clcache.py

```diff
@@ -817,29 +817,27 @@ def analyze(cmdline):
                 return AnalysisResult.MultipleSourceFilesComplex, None, None
             return AnalysisResult.MultipleSourceFilesSimple, sourceFiles, None
 
-        outputFile = None
-        if 'Fo' in options:
-            outputFile = options['Fo'][0]
-            outputFile = os.path.normpath(outputFile)
-
-            if os.path.isdir(outputFile):
-                outputFile = os.path.join(outputFile, basenameWithoutExtension(sourceFiles[0]) + ".obj")
-
-        elif preprocessing:
-            if 'P' in options:
-                # Preprocess to file.
+        if preprocessing:
+            if 'E' in options or 'EP' in options:
+                # Preprocess to stdout
+                outputFile = None
+            else:
+                # Preprocess to file
                 if 'Fi' in options:
                     outputFile = options['Fi'][0]
                 else:
                     outputFile = basenameWithoutExtension(sourceFiles[0]) + ".i"
-            else:
-                # Preprocess to stdout. Use empty string rather then None to ease
-                # output to log.
-                outputFile = ''
```

Comment:
Hm, I see commit a3771033 intentionally did not use `None` here, but I don't quite understand the comment. I suspect there was some string concatenation or so going on which `None` prohibits? In any case, this doesn't seem to hurt... I'm just wary of reverting code which has a comment attached to it explaining that things are done intentionally.